### PR TITLE
Refactor to treat different areas of the system as modules

### PR DESCRIPTION
**Pull request by jonnyshaw89**
### What is the context of this PR?
Have loaded the individual areas of the system in as modules.
This speeds up the creation and destruction of an environment by allowing infrastructure to be provisioned in parallel

A `developer_defaults.tf` file is defined in the root which reduces the amount of configuration required by developers to get an environment up and running

### How to review
Try and deploy an environment by following the README

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -46,19 +46,17 @@ brew install terraform
...
 
-1. Move the ssh key to `eq-terraform/survey-runner-queue`
+1. Move the ssh key to `eq-terraform`
 
-1. Copy `terraform.tfvars.example` to `terraform.tfvars` in both author and all survey-runner-* directories. You'll need to change all values containing an 'X' to match your requirements, including the AWS credentials you set up previously.
+1. Copy `terraform.tfvars.example` to `terraform.tfvars`. You'll need to change all values to match your requirements, including the AWS credentials you set up previously.
 
 1. To deploy survey-runner:
 
-  - Export an `AWS_ENVIRONMENT_NAME` environment variable e.g. `export AWS_ENVIRONMENT_NAME=preprod` or `export AWS_ENVIRONMENT_NAME=$USER`
```
Comment (ajmaddaford) [inline README.md:55]: You need to run a `terraform get` to import the modules before the plan. Probably worth mentioning here.

File: survey-runner-database/global_vars.tf [hunk 2]
```diff
@@ -21,6 +21,7 @@ variable "vpc_id" {
 }
 
 variable "private_route_table_ids" {
+  type = "list"
   description = "Route tables with route to NAT gateway"
```
Comment (ajmaddaford) [inline survey-runner-database/global_vars.tf:25]: When I run the `run-terraform.sh` script I get an error:
`variable private_route_table_ids should be type list, got string`

**Review by ajmaddaford — CHANGES_REQUESTED** [review]

File: survey-runner-database/global_vars.tf (hunk 2 above)
Comment (ajmaddaford) [reply survey-runner-database/global_vars.tf:25]: We need to change the scripts that pass lists to terraform to this format:
`-var 'private_route_table_ids=["<id_1>","<id_2>","<id_3>"]'`

**Review by ajmaddaford — CHANGES_REQUESTED** [review]

**Comment by jonnyshaw89** [comment]
@ajmaddaford i have addressed your comments.

File: survey-runner-application/run-terraform.sh [hunk 3]
```diff
@@ -35,7 +35,7 @@ if [ -z "$public_subnet_ids" ]; then
 fi
 
 # get the private route table ids as a comma separated list
-private_route_table_ids=`aws ec2 describe-route-tables --filters "Name=tag:Environment,Values=${AWS_ENVIRONMENT_NAME}" "Name=tag:Type,Values=Private" --query 'RouteTables[*].RouteTableId' --output text | tr '\t' ','`
+private_route_table_ids="[$(aws ec2 describe-route-tables --filters "Name=tag:Environment,Values=${AWS_ENVIRONMENT_NAME}" "Name=tag:Type,Values=Private" --query 'RouteTables[*].RouteTableId' --output text | tr '\t' ',' | sed -e 's/,/","/g')\"]"
```
Comment (ajmaddaford) [inline survey-runner-application/run-terraform.sh]: I get an error when running this:
```
invalid value "private_route_table_ids=[rtb-cc1d56ab\",\"rtb-ed13588a\",\"rtb-091c576e\"]"
```
Looks like a missing `"` before the first route table id

**Review by ajmaddaford — CHANGES_REQUESTED** [review]

**Comment by ajmaddaford** [comment]
I also get the error:
```
* variable public_subnet_ids should be type list, got string
```

**Review by ajmaddaford — APPROVED** [review]

**Review by samiwel — APPROVED** [review]
Worked for me, with the usual dose of Terraform/AWS fiddly-ness!

**Pull request merged**